Tests/Packet++Test/Tests/WireGuardTests.cpp

```diff
@@ -0,0 +1,336 @@
+	PTF_ASSERT_EQUAL(wgHandShakeResponseLayer->toString(),
+	                 "WireGuard Layer, " + wgHandShakeResponseLayer->getMessageTypeAsString() + " message");
+
+	PTF_ASSERT_TRUE(wgHandShakeResponseLayer->getWireGuardMessageType() ==
+	                pcpp::WireGuardLayer::WireGuardMessageType::HandshakeResponse);
+
+	PTF_ASSERT_EQUAL(wgHandShakeResponseLayer->getSenderIndex(), be32toh(2877158406));
+	PTF_ASSERT_EQUAL(wgHandShakeResponseLayer->getReceiverIndex(), be32toh(818952152));
+
+	std::array<uint8_t, 32> expectedResponderEphemeral = { 0xb1, 0x8d, 0x55, 0x50, 0xbd, 0x40, 0x42, 0xa3,
+		                                                   0x7a, 0x46, 0x82, 0x3a, 0xc0, 0x8d, 0xb1, 0xec,
+		                                                   0x66, 0x83, 0x9b, 0xc0, 0xca, 0x2d, 0x64, 0xbc,
+		                                                   0x15, 0xcd, 0x80, 0x23, 0x2b, 0x66, 0x23, 0x2f };
+	PTF_ASSERT_TRUE(wgHandShakeResponseLayer->getResponderEphemeral() == expectedResponderEphemeral);
+
+	std::array<uint8_t, 16> encryptedEmptyData = { 0xae, 0xc2, 0x4a, 0xf8, 0x91, 0x8d, 0xe1, 0x06,
+		                                           0x0f, 0xf5, 0xc9, 0x8e, 0x86, 0x5d, 0x5f, 0x35 };
+
+	PTF_ASSERT_TRUE(wgHandShakeResponseLayer->getEncryptedEmpty() == encryptedEmptyData);
+
+	std::array<uint8_t, 16> expectedMac1 = { 0xf2, 0x72, 0x21, 0x4c, 0x52, 0x60, 0x11, 0x0d,
+		                                     0xc4, 0xc6, 0x1e, 0x32, 0xcd, 0xd8, 0x54, 0x21 };
+	PTF_ASSERT_TRUE(wgHandShakeResponseLayer->getMac1() == expectedMac1);
+
+	std::array<uint8_t, 16> expectedMac2 = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		                                     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+	PTF_ASSERT_TRUE(wgHandShakeResponseLayer->getMac2() == expectedMac2);
+}
+
+PTF_TEST_CASE(WireGuardCookieReplyParsingTest)
+{
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/WireGuardCookieReply.dat");
+
+	pcpp::Packet wgCookieReplyPacket(&rawPacket1);
+
+	PTF_ASSERT_TRUE(wgCookieReplyPacket.isPacketOfType(pcpp::WireGuard));
+	pcpp::WireGuardLayer* wgLayer = wgCookieReplyPacket.getLayerOfType<pcpp::WireGuardLayer>();
+	PTF_ASSERT_NOT_NULL(wgLayer);
+
+	pcpp::WireGuardCookieReplyLayer* wgCookieReplyaLayer =
+	    wgCookieReplyPacket.getLayerOfType<pcpp::WireGuardCookieReplyLayer>();
+	PTF_ASSERT_NOT_NULL(wgCookieReplyaLayer);
+
+	PTF_ASSERT_EQUAL(wgCookieReplyaLayer->getMessageTypeAsString(), "Cookie Reply");
+	PTF_ASSERT_EQUAL(wgCookieReplyaLayer->toString(),
+	                 "WireGuard Layer, " + wgCookieReplyaLayer->getMessageTypeAsString() + " message");
+
+	PTF_ASSERT_TRUE(wgCookieReplyaLayer->getWireGuardMessageType() ==
+	                pcpp::WireGuardLayer::WireGuardMessageType::CookieReply);
+
+	uint32_t receiverIndex = 2877158406;
+
+	PTF_ASSERT_EQUAL(wgCookieReplyaLayer->getReceiverIndex(), receiverIndex);
```

Comment:
We can just do: `PTF_ASSERT_EQUAL(wgCookieReplyaLayer->getReceiverIndex(), 2877158406);`